finetuner/models.py

```diff
@@ -0,0 +1,158 @@
+import abc
+from typing import Any, Dict, List, Optional, Tuple, TypeVar
+
+ModelStubType = TypeVar('ModelStubType', bound='_ModelStub')
+
+
+class _ModelStub(metaclass=abc.ABCMeta):
+    name: str
+    description: str
+    task: str
+    output_dim: Optional[int]
+    architecture: str
+    options: Dict[str, Any]
+
+    def __init__(self, **kwargs):
+        self.options = kwargs
+
+    @staticmethod
+    def header() -> Tuple[str, ...]:
+        """Get table header."""
+        return 'name', 'task', 'output_dim', 'architecture', 'description'
+
+    @classmethod
+    def row(cls) -> Tuple[str, ...]:
+        """Get table row."""
+        return (
+            cls.name,
+            cls.task,
+            str(cls.output_dim),
+            cls.architecture,
+            cls.description,
+        )
+
+
+class MLP(_ModelStub):
+    """MLP model stub.
+
+    :param input_size: Size of the input representations.
+    :param hidden_sizes: A list of sizes of the hidden layers. The last hidden size is
+        the output size.
+    :param bias: Whether to add bias to each layer.
+    :param activation: A string to configure activation function, `relu`, `tanh` or
+        `sigmoid`. Set to `None` for no activation.
+    :param l2: Apply L2 normalization at the output layer.
+    """
+
+    name = 'mlp'
+    description = 'Simple MLP encoder trained from scratch'
+    task = 'any'
+    output_dim = '-'
+    architecture = 'MLP'
+
+    def __init__(
+        self,
+        input_size: int,
+        hidden_sizes: Optional[List[int]] = None,
+        bias: bool = True,
+        activation: Optional[str] = None,
+        l2: bool = False,
+    ):
+        super(MLP, self).__init__(
+            input_size=input_size,
+            hidden_sizes=hidden_sizes,
+            bias=bias,
+            activation=activation,
+            l2=l2,
+        )
+
+
+class ResNet50(_ModelStub):
+    """ResNet50 model stub."""
+
+    name = 'resnet50'
+    description = 'Pretrained on ImageNet'
+    task = 'image-to-image'
+    output_dim = '2048'
+    architecture = 'CNN'
+
+    def __init__(self):
+        super(ResNet50, self).__init__()
```

Comment:
can we unify the coding style as MLP? `super().__init__(name=name, ...)`

Reply:
what do you mean?